src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/CodeView/CodeViewSymbolsBuilder.cs

```diff
@@ -0,0 +1,432 @@
+                                {
+                                    recordWriter.Write((ushort)cvRegNum);
+                                    // TODO: Flags, CV_OFFSET_PARENT_LENGTH_LIMIT
+                                    recordWriter.Write((ushort)0);
+                                    recordWriter.Write((uint)range.VarLoc.C);
+                                    recordWriter.EmitSymbolReference(RelocType.IMAGE_REL_SECREL, methodName, (int)range.StartOffset);
+                                    recordWriter.EmitSymbolReference(RelocType.IMAGE_REL_SECTION, methodName);
+                                    recordWriter.Write((ushort)rangeLength);
+                                }
+                            }
+                            break;
+
+                        case VarLocType.VLT_REG_BYREF:
+                        case VarLocType.VLT_STK_BYREF:
+                        case VarLocType.VLT_REG_REG:
+                        case VarLocType.VLT_REG_STK:
+                        case VarLocType.VLT_STK_REG:
+                        case VarLocType.VLT_STK2:
+                        case VarLocType.VLT_FPSTK:
+                        case VarLocType.VLT_FIXED_VA:
+                            break;
+
+                        default:
+                            Debug.Fail("Unknown variable location type");
+                            break;
+                    }
+                }
+            }
+
+            using (var recordWriter = symbolSubsection.StartRecord(S_PROC_ID_END))
+            {
+            }
+        }
+
+        public void EmitLineInfo(
+            CodeViewFileTableBuilder fileTableBuilder,
+            string methodName,
+            int methodPCLength,
+            IEnumerable<NativeSequencePoint> sequencePoints)
+        {
+            using var lineSubsection = GetSubsection(DebugSymbolsSubsectionType.Lines);
+
+            using (var recordWriter = lineSubsection.StartRecord())
+            {
+                recordWriter.EmitSymbolReference(RelocType.IMAGE_REL_SECREL, methodName);
+                recordWriter.EmitSymbolReference(RelocType.IMAGE_REL_SECTION, methodName);
+                recordWriter.Write((ushort)0); // TODO: Flags (eg. have columns)
+                recordWriter.Write((uint)methodPCLength);
+
+                string lastFileName = null;
+                uint fileIndex = 0;
+                List<uint> codes = new();
+
+                foreach (var sequencePoint in sequencePoints)
+                {
+                    if (lastFileName is null || lastFileName != sequencePoint.FileName)
+                    {
+                        if (codes.Count > 0)
+                        {
+                            recordWriter.Write(fileIndex);
+                            recordWriter.Write((uint)(codes.Count / 2));
+                            recordWriter.Write((uint)(12 + 4 * codes.Count));
```

Comment:
Yep, deserves a comment. I'll double check with my notes and the specs. IIRC it's the full record size (3 `uint`s for file index, `offset`+`code` item count, and record size incl. the header size). The `/ 2` part is awkward because the codes are pairs of `offset`+`code` and this specifies the number of the pairs while we store them as `uint`s in a single list.